ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2StorageBackend.scala

```diff
@@ -214,4 +214,37 @@ private[backend] object H2StorageBackend
     throw new UnsupportedOperationException("db level locks are not supported for H2")
 
   override def dbLockSupported: Boolean = false
+
+  // Events
+  override def pruneImmediateDivulgence(
+      pruneUpToInclusive: Offset,
+      pruneAllDivulgedContracts: Boolean,
+  ): List[SimpleSql[Row]] = {
+    import com.daml.platform.store.Conversions.OffsetToStatement
+    if (pruneAllDivulgedContracts) {
+      List(
+        SQL"""
+          -- Immediate divulgence events
+          WITH to_be_pruned AS (
+            SELECT DISTINCT(event_sequential_id) AS event_sequential_id
+            FROM participant_events_create creates
+            WHERE event_offset <= $pruneUpToInclusive
+            AND NOT EXISTS (
+              SELECT 1
+              FROM participant_events_create c JOIN parties p
+              ON array_contains(c.flat_event_witnesses, p.party)
+              WHERE
+                c.event_sequential_id = creates.event_sequential_id
+                -- If the party is locally hosted and older than the create event
+                -- it is not a prunable disclosure
+                AND p.is_local
+                AND p.ledger_offset <= c.event_offset
+            )
```

Comment:
As above: inline this condition directly instead of going indirectly via `event_sequential_id`.

Reply:
and consider whether you can use unnest as well here as in Postgres. I suspect it will have better performance, as it will direct the query planner to do prepare a hash-table for the party table.